### ISSUE TITLE
Librosa not compatible with numpy complex numbers

### ISSUE THREAD

**Issue by Shivamkak19** [librosa/librosa#1853]
**Title:** Deprecation Warning: Use of `np.complex` in `constantq.py` Causes AttributeError with NumPy 2.0.0

**Description:**

The `librosa` library is currently using `np.complex` in `constantq.py`, which causes an `AttributeError` when using NumPy version 2.0.0 or later. The use of `np.complex` has been deprecated since NumPy 1.20.0 and removed in NumPy 2.0.0.

**Error Message:**
```python
Traceback (most recent call last):
  File "/path/to/your/script.py", line 1, in <module>
    import librosa
  File "/path/to/librosa/__init__.py", line 211, in <module>
    from . import core
  File "/path/to/librosa/core/__init__.py", line 9, in <module>
    from .constantq import *  # pylint: disable=wildcard-import
  File "/path/to/librosa/core/constantq.py", line 1059, in <module>
    dtype=np.complex,
          ^^^^^^^^^^
  File "/path/to/numpy/__init__.py", line 394, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'complex'.

**Suggested Fix**
complex(1)         # built-in complex type
# or
np.complex128(1)   # specific NumPy scalar type
# or
np.complex_(1)     # NumPy alias
# or
np.cdouble(1)      # another NumPy alias

**Comment by bmcfee** [comment]
Thanks for noting this.  We're currently blocked on numpy 2.0 support, see #1831 for status tracking of our upstream dependencies.  I'm tagging this for the 0.11 release.

As an aside, this will take a full package audit to smooth over (we use complex in many places), and may need to do some type annotation revisions to get it all working.

**Comment by bmcfee** [comment]
@Shivamkak19 actually, would you mind filling out the issue form with your environment details?  I think you may be on an old version, and the type deprecations should have already been resolved when we fixed #1350 and #1390.

**Comment by bmcfee** [comment]
Confirming that this was closed several years ago in the 0.9 release.
